flow/project.py

```diff
@@ -400,7 +396,9 @@ def __init__(self, id, name, job, cmd, directives=None):
         self._cmd = cmd
 
         if directives is None:
-            directives = dict()  # default argument
+            base_directives = job._project._environment._get_default_directives()
+            base_directives.update(directives)
+            directives = base_directives
```

Comment:
This looks like it's performing `base_directives.update(None)`? `dict.update(None)` doesn't even work in Python...?
```suggestion
            directives = job._project._environment._get_default_directives()
```